src/api/goods.ts

```
@@ -1,16 +1,37 @@
 import { Good } from '../types/Good';
 
-// eslint-disable-next-line
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
-export function getAll(): Promise<Good[]> {
-  return fetch(API_URL).then(response => response.json());
+export async function getAllGoods(): Promise<Good[]> {
+  try {
+    const response = await fetch(API_URL);
+
+    if (!response.ok) {
+      throw new Error(`Error: ${response.status}`);
+    }
+
+    return await response.json();
+  } catch (error) {
+    throw new Error('Failed to load goods');
+  }
+}
+
+export async function getFirstFiveGoods(): Promise<Good[]> {
+  try {
+    const goods = await getAllGoods();
+
+    return [...goods].sort((a, b) => a.name.localeCompare(b.name)).slice(0, 5);
+  } catch (error) {
+    throw new Error('Failed to load 5 first goods:');
+  }
 }
 
-export const get5First = () => {
-  return getAll().then(goods => goods); // sort and get the first 5
-};
+export async function getRedGoods(): Promise<Good[]> {
+  try {
+    const goods = await getAllGoods();
 
-export const getRedGoods = () => {
-  return getAll().then(goods => goods); // get only red
-};
+    return goods.filter(good => good.color === 'red');
+  } catch (error) {
+    throw new Error('Failed to load red goods:');
```

Similar to the other function, this error message discards the original error's details. Propagating the original error would make it easier to trace the source of the problem.